### DISCUSSION TITLE
How to draw Text with a background color?

### DISCUSSION THREAD

**Discussion by jmaister** [hajimehoshi/ebiten#1717] [Q&A] [answered]
text.Draw has a color for the foreground, but not for the background:

    text.Draw(screen, char, font, x, y, foregroundColor)

**Comment by hajimehoshi** [comment, marked as answer]
I'm not sure what the background color of a text means. For shadows, you can render it by calling text.Draw multiple times.

**Reply by jmaister** [reply]
For background, I mean the rectangle under the character or characters that have been drawn.

For example:
![imagen](https://user-images.githubusercontent.com/374021/125845971-2c890335-9861-4815-a742-3accc51faa59.png)

**Reply by hajimehoshi** [reply]
You can get the rectangle by [`text.BoundString`](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2/text#BoundString). You can fill the background by yourself by [`ebitenutil.DrawRect`](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2/ebitenutil#DrawRect).

**Reply by jmaister** [reply]
Nice! Thank you, it works.

![image](https://user-images.githubusercontent.com/374021/125978056-e97d1e2f-8864-4aa0-a6ad-fa3800d89dc3.png)

Here is the code:

```go
func drawBackground2(screen *ebiten.Image, bgColor color.Color, x int, y int, face font.Face, str string) {
	rect := text.BoundString(face, str)
	ebitenutil.DrawRect(screen, float64(x+rect.Min.X), float64(y+rect.Min.Y), float64(rect.Max.X-rect.Min.X), float64(rect.Max.Y-rect.Min.Y), bgColor)
}
```

**Reply by kendfss** [reply]
> You can get the rectangle by [`text.BoundString`](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2/text#BoundString). You can fill the background by yourself by [`ebitenutil.DrawRect`](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2/ebitenutil#DrawRect).

Is this still the recommended approach? There doesn't seem to be a BoundString function in `text/v2`.

**Reply by hajimehoshi** [reply]
You can use [Measure](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2/text/v2#Measure).
